financial-connections/src/main/java/com/stripe/android/financialconnections/domain/GetOrFetchSync.kt

```diff
@@ -17,10 +17,44 @@ internal class GetOrFetchSync @Inject constructor(
     @Named(APPLICATION_ID) private val applicationId: String,
 ) {
 
-    suspend operator fun invoke(): SynchronizeSessionResponse {
-        return repository.getOrFetchSynchronizeFinancialConnectionsSession(
+    suspend operator fun invoke(
+        fetchCondition: FetchCondition = FetchCondition.IfMissing
+    ): SynchronizeSessionResponse {
+        return repository.getOrSynchronizeFinancialConnectionsSession(
             clientSecret = configuration.financialConnectionsSessionClientSecret,
-            applicationId = applicationId
+            applicationId = applicationId,
+            fetchCondition = fetchCondition::check,
         )
     }
+
+    sealed interface FetchCondition {
+        fun check(response: SynchronizeSessionResponse): Boolean
```

Comment:
Nit: The naming doesn’t tell what the result indicates. Can we rename this to `shouldForceRefresh` or `shouldLoadFromBackend` or something like that?

This also makes it easier to understand at the call site:

```kotlin
cachedSync.takeUnless { shouldLoadFromBackend(it) }
```